### PR TITLE
Maesemen/stiffstiff snes

### PR DESCRIPTION
**Pull request by maesemen**
Uses iterative solve to use VoF with two liquids using stiffened gas EOS.  Changes order of analytical solution for gas-stiff for rhoG first to decrease computational error in solution.  Allow for negative pressure in liquid-liquid interface, feature of stiffened gas EOS.

### CODE REVIEW

File: src/finiteVolume/fluxCalculator/riemannStiff.cpp [hunk 1]
```diff
@@ -89,12 +89,17 @@ ablate::finiteVolume::fluxCalculator::Direction ablate::finiteVolume::fluxCalcul
 
     // iteration starts
     while (PetscAbsReal(f_L_0 + f_R_0 + del_u) > err && i <= MAXIT)  // Newton's method
+                                                                     //    while (2* PetscAbs((pstar - pold)/(pstar + pold))> err && i <=MAXIT)
```
Comment (mmcgurn) [inline src/finiteVolume/fluxCalculator/riemannStiff.cpp:92]: Would you mind removing dead code?

File: src/finiteVolume/processes/twoPhaseEulerAdvection.cpp [hunk 2]
```diff
@@ -67,9 +232,16 @@ PetscErrorCode ablate::finiteVolume::processes::TwoPhaseEulerAdvection::Compress
     PetscFunctionBeginUser;
     auto twoPhaseEulerAdvection = (TwoPhaseEulerAdvection *)ctx;
     // Compute the norm of cell face
-    PetscReal norm[3];
+    PetscReal norm[3], newnormal[3];
```
Comment (mmcgurn) [inline src/finiteVolume/processes/twoPhaseEulerAdvection.cpp:235]: Is this needed?

File: tests/unitTests/finiteVolume/processes/twoPhaseEulerAdvectionTests.cpp [hunk 3]
```diff
@@ -181,19 +181,19 @@ INSTANTIATE_TEST_SUITE_P(TwoPhaseEulerAdvectionTests, TwoPhaseEulerAdvectionTest
                                  .conservedValuesIn = {340.4, 851000000.0, 3404.0, -6808.0, 10212.0, 0.8},  // RHO, RHOE, RHOU, RHOV, RHOW, RHOALPHA
                                  .normalIn = {0.5, 0.5, 0.7071},                                            // x, y, z
                                  .expectedDensity = 340.4,
-                                 .expectedDensityG = 1.2352213315429057,
-                                 .expectedDensityL = 963.834108739261,
+                                 .expectedDensityG = 1.2352213315429055,
```
Comment (mmcgurn) [inline tests/unitTests/finiteVolume/processes/twoPhaseEulerAdvectionTests.cpp:184]: Do any of these tests test StiffenedGasStiffenedGasDecoder?

**Review by mmcgurn — CHANGES_REQUESTED** [review]
Looks good, I just had some questions?

File: src/finiteVolume/processes/twoPhaseEulerAdvection.cpp (hunk 2 above)
Comment (maesemen) [reply src/finiteVolume/processes/twoPhaseEulerAdvection.cpp:235]: Good point, I don't think this changed solved anything, so I will get rid of this

File: tests/unitTests/finiteVolume/processes/twoPhaseEulerAdvectionTests.cpp (hunk 3 above)
Comment (maesemen) [reply tests/unitTests/finiteVolume/processes/twoPhaseEulerAdvectionTests.cpp:184]: No, I will add some tests for that

**Review by mmcgurn — APPROVED** [review]
Thanks!

**Pull request merged**